ksql-engine/src/test/java/io/confluent/ksql/util/KafkaTopicClientImplTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import io.confluent.ksql.testutils.EmbeddedSingleNodeKafkaCluster;
+
+import static io.confluent.ksql.testutils.AssertEventually.assertThatEventually;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.is;
+
+@Category({IntegrationTest.class})
+public class KafkaTopicClientImplTest {
+  @ClassRule
+  public static final EmbeddedSingleNodeKafkaCluster KAFKA =
+      EmbeddedSingleNodeKafkaCluster.newBuilder().build();
+
+  private String testTopic;
+  private KafkaTopicClient client;
+  private AdminClient adminClient;
+
+  @Before
+  public void setUp() {
+    testTopic = UUID.randomUUID().toString();
+    KAFKA.createTopic(testTopic);
+
+    adminClient = AdminClient.create(ImmutableMap.of(
+        AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, KAFKA.bootstrapServers()));
+
+    client = new KafkaTopicClientImpl(adminClient);
+  }
+
+  @After
+  public void tearDown() {
+    adminClient.close();
+  }
+
+  @Test
+  public void shouldGetTopicConfig() {
+    // When:
+    final Map<String, String> config = client.getTopicConfig(testTopic);
+
+    // Then:
+    assertThat(config.keySet(), hasItems(
+        TopicConfig.RETENTION_MS_CONFIG,
+        TopicConfig.CLEANUP_POLICY_CONFIG,
+        TopicConfig.COMPRESSION_TYPE_CONFIG));
+  }
+
+  @Test
+  public void shouldSetTopicConfig() {
+    // When:
+    final boolean changed =
+        client.addTopicConfig(testTopic, ImmutableMap.of(TopicConfig.RETENTION_MS_CONFIG, "1245678"));
+
+    // Then:
+    assertThat(changed, is(true));
+    assertThatEventually(() -> getTopicConfig(TopicConfig.RETENTION_MS_CONFIG), is("1245678"));
+  }
+
+  @Test
+  public void shouldSetTopicConfigWhenNothingChanged() {
```

Comment:
should this be named shouldNotSetTopicConfigWhenNothingChanged?

Reply:
Ermm... good spot!